bin/migrate-oats-data/noi/noi_decisions/noi_components/notice_of_intent_decision_components_init.py

```diff
@@ -0,0 +1,199 @@
+                    ag_cap_consultant,
+                    ag_cap_map,
+                    ag_cap_source,
+                    alr_area,
+                    audit_created_by,
+                    end_date,
+                    end_date2,
+                    expiry_date,
+                    notice_of_intent_decision_component_type_code,
+                    notice_of_intent_decision_uuid,
+                    oats_alr_appl_component_id
+                )
+                VALUES (
+                    %(ag_cap)s,
+                    %(ag_cap_consultant)s,
+                    %(ag_cap_map)s,
+                    %(ag_cap_source)s,
+                    %(alr_area)s,
+                    %(audit_created_by)s,
+                    %(end_date)s,
+                    %(end_date2)s,
+                    %(expiry_date)s,
+                    %(notice_of_intent_decision_component_type_code)s,
+                    %(notice_of_intent_decision_uuid)s,
+                    %(alr_appl_component_id)s
+                )
+                ON CONFLICT DO NOTHING; -- there are no components on prod
+    """
+    return query
+
+
+def _prepare_oats_alr_applications_data(row_data_list):
+    data_list = []
+    for row in row_data_list:
+        if row.get("alr_change_code", None) in [
+            ALRChangeCode.SCH.value,
+            ALRChangeCode.FILL.value,
+            ALRChangeCode.EXT.value,
+        ]:
+            mapped_row = {
+                "ag_cap": str(OatsToAlcsAgCap[row["agri_capability_code"]].value)
+                if row.get("agri_capability_code")
+                else None,
+                "ag_cap_consultant": row.get("agri_cap_consultant"),
+                "ag_cap_map": row.get("agri_cap_map"),
+                "ag_cap_source": str(
+                    OatsToAlcsAgCapSource[row["capability_source_code"]].value
+                )
+                if row.get("capability_source_code")
+                else None,
+                "alr_area": row.get("component_area"),
+                "audit_created_by": OATS_ETL_USER,
+                "end_date": add_timezone_and_keep_date_part(
+                    row.get("nonfarm_use_end_date")
+                ),
+                "end_date2": add_timezone_and_keep_date_part(
+                    row.get("nonfarm_use_end_date")
+                ),
+                "expiry_date": add_timezone_and_keep_date_part(
+                    row.get("decision_expiry_date")
+                ),
+                "notice_of_intent_decision_component_type_code": _map_component_type_code(
+                    row
+                ),
+                "notice_of_intent_decision_uuid": row.get("decision_uuid"),
+                "alr_appl_component_id": row.get("component_id"),
+            }
+            data_list.append(mapped_row)
+
+    return data_list
+
+
+def _map_component_type_code(row):
+    alr_change_code = row.get("alr_change_code")
+    if alr_change_code == ALRChangeCode.SCH.value:
+        return "PFRS"
```

Comment:
Future note to change this to enum when doing applications

Reply:
agree for applications since there will be more than 3 values and it will refer to different table in DB